nautobot/core/filters.py

```diff
@@ -716,8 +715,23 @@ def get_filters(cls):
         """
         filters = super().get_filters()
 
+        if "static_groups" not in filters and getattr(cls._meta.model, "is_static_group_associable_model", False):
+            # Add "static_groups" field as the last key
+            from nautobot.extras.models import StaticGroup
+
+            filters["static_groups"] = NaturalKeyOrPKMultipleChoiceFilter(
+                queryset=StaticGroup.objects.all(),
+                field_name="associated_static_groups__static_group",
+                to_field_name="name",
+                query_params={"content_type": cls._meta.model._meta.label_lower},
+                label="Static groups (name or ID)",
+            )
+
         # django-filters has no concept of "abstract" filtersets, so we have to fake it
         if cls._meta.model is not None:
+            if "tags" in filters and isinstance(filters["tags"], TagFilter):
+                filters["tags"].extra["query_params"] = {"content_types": [cls._meta.model._meta.label_lower]}
+
```

Comment:
This is part of the logic I moved out of `get_filterset_parameter_form_field` to be more generally applicable.